benchmark-service/benchmark/benchmark.py

```diff
@@ -214,26 +217,34 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'compare.html', context)
 
 
-async def query_github(app):
-    global START_POINT
-    github_client = app['github_client']
-    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+@router.post('/api/v1alpha/benchmark/create_benchmark')
+async def submit(request):  # pylint: disable=unused-argument
+    app = request.app
+    batch_client = app['batch_client']
+    body = await request.json()
+    commit = body['commit']
+    print(commit)
+    batch_id = await submit_batch(commit, batch_client)
+    return web.HTTPFound(
+        deploy_config.external_url('benchmark', f'/api/v1alpha/benchmark/batches/{batch_id}'))
+
 
-    data = await github_client.getitem(request_string)
-    new_commits = []
-    for commit in data:
-        sha = commit.get('sha')
-        new_commits.append(commit)
-        log.info(f'commit {sha}')
-        START_POINT = commit['commit']['author'].get('date')
-        log.info(f'start point is now {START_POINT}')
-    log.info('got new commits')
+@router.get('/api/v1alpha/benchmark/batches/{batch_id}')
+async def batch_status(request):  # pylint: disable=unused-argument
+    app = request.app
```

Comment:
I'm not sure what your reasoning is for this interface. I think it should take the sha to look up and then query for batches that match that sha. Then return the status for the last batch that matches the sha (I think that's the correct ordering) or returns None if it doesn't exist.